spyder/config/user.py

```diff
@@ -5,369 +5,498 @@
-        """Create a .ini filename located in the current project directory.
-        This .ini files stores the specific project preferences for each
-        project created with spyder.
-        """
-        return osp.join(self._root_path, self._filename)
-
-    def _filename_global(self):
-        """Create a .ini filename located in user home directory.
-        This .ini files stores the global spyder preferences.
-        """
-        if self.subfolder is None:
-            config_file = osp.join(get_home_dir(), '.%s.ini' % self.name)
-            return config_file
-        else:
-            folder = get_conf_path()
-            # Save defaults in a "defaults" dir of .spyder2 to not pollute it
-            if 'defaults' in self.name:
-                folder = osp.join(folder, 'defaults')
-                if not osp.isdir(folder):
-                    os.mkdir(folder)
-            config_file = osp.join(folder, '%s.ini' % self.name)
-            return config_file
+    def get_config_fpath(self):
+        """Return the ini file where this configuration is stored."""
+        path = self._path
+        config_file = osp.join(path, '{}.ini'.format(self._name))
+        return config_file
 
     def set_defaults(self, defaults):
+        """Set default values and save to defaults folder location."""
         for section, options in defaults:
             for option in options:
-                new_value = options[ option ]
-                self._set(section, option, new_value, False)
+                new_value = options[option]
+                self._set(section, option, new_value, verbose=False)
 
 
-#==============================================================================
+# ============================================================================
 # User config class
-#==============================================================================
+# ============================================================================
 class UserConfig(DefaultsConfig):
     """
-    UserConfig class, based on ConfigParser
-    name: name of the config
-    defaults: dictionnary containing options
-              *or* list of tuples (section_name, options)
-    version: version of the configuration file (X.Y.Z format)
-    subfolder: configuration file will be saved in %home%/subfolder/%name%.ini
-
-    Note that 'get' and 'set' arguments number and type
-    differ from the overriden methods
+    UserConfig class, based on ConfigParser.
+
+    Parameters
+    ----------
+    name: str
+        Name of the config
+    path: str
+        Configuration file will be saved in path/%name%.ini
+    defaults: {} or [(str, {}),]
+        dictionnary containing options *or* list of tuples (sec_name, options)
+    load: bool
+        TODO:
+    version: str
+        version of the configuration file in 'major.minor.micro' format.
+    backup: bool
+        TODO:
+    raw_mode: bool
+        TODO:
+    remove_obsolete: bool
+        TODO:

```

Comment:
Please fill the TODOs you left in this docstring.

Reply:
Done!